### PR TITLE
Improve Fixity Result response

### PR DESCRIPTION
**Pull request by claussni**
https://jira.duraspace.org/browse/FCREPO-2297

### CODE REVIEW

File: fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/RdfLexicon.java [hunk 1]
```diff
@@ -84,8 +84,13 @@
     // FIXITY
 
     public static final Resource FIXITY_TYPE = createResource(PREMIS_NAMESPACE + "Fixity");
+
+    public static final Property HAS_MESSAGE_DIGEST_ALGORITHM =
+            createProperty(PREMIS_NAMESPACE + "hasMessageDigestAlgorithm");
```
Comment (awoods) [inline fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/RdfLexicon.java]: I am not seeing the `hasMessageDigestAlgorithm` in the Premis namespace:
http://www.loc.gov/standards/premis/v1/premis-dd_1.0_2005_May.pdf

Comment (claussni) [inline fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/RdfLexicon.java]: Here is one: http://id.loc.gov/ontologies/premis.html#p_hasMessageDigestAlgorithm
There is also a `messageDigestAlgorithm` on page 25 of the mentioned PDF.
Maybe this is another Premis version?

Comment (awoods) [inline fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/RdfLexicon.java]: Thanks.

Comment (escowles) [inline fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/RdfLexicon.java]: FYI, the current authoritative PREMIS ontology is http://id.loc.gov/ontologies/premis.rdf (linked to from the bottom of the HTML page @claussni linked to) and the current version of the data dictionary reference is http://www.loc.gov/standards/premis/v3/premis-3-0-final.pdf.

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraFixity.java [hunk 2]
```diff
@@ -92,11 +92,12 @@ public RdfNamespacedStream getDatastreamFixity() {
         if (!(resource() instanceof FedoraBinary)) {
             throw new NotFoundException(resource() + " is not a binary");
         }
+        final FedoraBinary binary = (FedoraBinary) resource();
 
         LOGGER.info("Get fixity for '{}'", externalPath);
         return new RdfNamespacedStream(
                 new DefaultRdfStream(asNode(resource()),
-                    ((FedoraBinary)resource()).getFixity(translator())),
+                    binary.getFixity(translator())),
```
Comment (awoods) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraFixity.java]: Whenever possible, it is helpful to minimize combining stylistic/non-functional changes within a functional pull-request.
I would suggest reverting the changes in this file, as they do not appear to serve any functional purpose... unless I am missing something.

File: fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/FixityResultImpl.java [hunk 3]
```diff
@@ -91,16 +92,19 @@ public String getStoreIdentifier() {
...
     @Override
     public boolean equals(final Object obj) {
-
-        boolean result = false;
         if (obj instanceof FixityResult) {
             final FixityResult that = (FixityResult) obj;
-            result =
-                    computedSize == that.getComputedSize() &&
-                            computedChecksum.equals(that.getComputedChecksum());
+            return computedSize == that.getComputedSize() &&
```
Comment (awoods) [inline fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/FixityResultImpl.java]: Please add a minor update the the following unit test:
https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-modeshape/src/test/java/org/fcrepo/kernel/modeshape/utils/FixityResultImplTest.java#L38

**Pull request merged**